ouroboros-consensus/test-storage/Test/Ouroboros/Storage/VolatileDB/Model.hs

```diff
@@ -1,109 +1,325 @@
-openDBModel err = (dbModel, db)
+openDBModel err maxNumPerFile toSlot lenientParsing = (dbModel, db)
     where
-        dbModel = initDBModel
-        db = VolatileDB {
+        dbModel = initDBModel maxNumPerFile lenientParsing
+        db =  VolatileDB {
               closeDB        = closeDBModel
             , isOpenDB       = isOpenModel
-            , reOpenDB       = reOpenModel
+            , reOpenDB       = reOpenModel err
             , getBlock       = getBlockModel err
-            , putBlock       = putBlockModel err
+            , putBlock       = putBlockModel err maxNumPerFile toSlot
             , garbageCollect = garbageCollectModel err
             , getIsMember    = getIsMemberModel err
         }
 
-closeDBModel :: MonadState (DBModel blockId) m => m ()
+closeDBModel :: MonadState (MyState blockId) m => m ()
 closeDBModel = do
-    dbm <- get
-    put $ dbm {open = False}
+    dbm <- getDB
+    putDB $ dbm {open = False}
 
-isOpenModel :: MonadState (DBModel blockId) m => m Bool
+isOpenModel :: MonadState (MyState blockId) m => m Bool
 isOpenModel = do
-    DBModel {..} <- get
+    DBModel {..} <- getDB
     return open
 
-reOpenModel :: MonadState (DBModel blockId) m => m ()
-reOpenModel = do
-    dbm <- get
-    put $ dbm {open = True}
+reOpenModel :: MonadState (MyState blockId) m
+            => ErrorHandling (VolatileDBError blockId) m
+            -> m ()
+reOpenModel err = do
+    dbm <- getDB
+    dbm' <- if not $ open dbm
+            then recover err dbm
+            else return dbm
+    putDB dbm' {open = True}
 
-getBlockModel :: forall m blockId. (MonadState (DBModel blockId) m, Ord blockId)
+getBlockModel :: forall m blockId. (MonadState (MyState blockId) m, Ord blockId)
               => ErrorHandling (VolatileDBError blockId) m
               -> blockId
               -> m (Maybe ByteString)
 getBlockModel err sl = do
-    DBModel {..} <- get
+    DBModel {..} <- getDB
     if not open then EH.throwError err ClosedDBError
     else return $ Map.lookup sl mp
 
-putBlockModel :: MonadState (DBModel blockId) m
+putBlockModel :: MonadState (MyState blockId) m
               => Ord blockId
               => ErrorHandling (VolatileDBError blockId) m
+              -> Int
+              -> (blockId -> Slot)
               -> blockId
               -> Builder
               -> m ()
-putBlockModel err sl bs = do
-    dbm@DBModel {..} <- get
+putBlockModel err maxNumPerFile toSlot bid bs = do
+    let managesToPut errors = do
+            errs <- errors
+            (mErr, _rest) <- uncons $ getStream (_hPut errs)
+            (fsErr, _mCorr) <- mErr
+            return fsErr
+    (dbm@DBModel {..}, cmdErr) <- get
     if not open then EH.throwError err ClosedDBError
-    else case Map.lookup sl mp of
-        Nothing -> put dbm{mp = Map.insert sl (toStrict $ toLazyByteString bs) mp}
+    else case Map.lookup bid mp of
         Just _bs -> return ()
+        Nothing -> case managesToPut cmdErr of
```

Comment:
Let's add a comment here that _if_ the exact sequence of file operations changes in the real implementation, this test logic would need to be updated, _or_ the test would have to be made less specific.